lib/game.rb

```diff
@@ -56,8 +49,8 @@ def second_player(players)
   end
 
   def opponent(ui)
-    opponent_choice = ui.choose_opponent
-    OpponentFactory.new(ui).create_opponent(opponent_choice)
+    @opponent_choice = ui.choose_opponent
+    OpponentFactory.new(ui).create_opponent(@opponent_choice)
```

Comment:
Why using `@` here?

I am not sure why you would inject the UI in the factory
Wouldn't it be easier with something like:

```
opponent_selected = ui.choose_opponent
OpponentFactory.create_opponent(opponent_selected)
```

Reply:
You could even inline that and not use the variable `opponent_choice` at all